client/src/components/Game/Timer.js

```diff
@@ -0,0 +1,136 @@
+            if(snapshot.data().currentPlayerIndex === 0) {
+              this.setState({
+                startTime: snapshot.data().gameStartTime.seconds,
+                timePerTurnInSeconds: parseInt(snapshot.data().timeLimit, 10) * 60,
+              })
+            } else {
+              this.setState({
+                startTime: Math.floor(new Date().getTime() / 1000),
+                timePerTurnInSeconds: parseInt(snapshot.data().timeLimit, 10) * 60,
+              })
+            }
+            this.setState({
+              timeTurnWillEnd: (this.state.timePerTurnInSeconds + this.state.startTime),
+            })
+            this.setState({
+              days: this.calculateTime('days'),
+              hours: this.calculateTime('hours'),
+              minutes: this.calculateTime('minutes'),
+              seconds: this.calculateTime('seconds')
+            })
+          }
+        })
+      }
+    })
+
+    this.myInterval = setInterval(async () => {
+      const { days, hours, minutes, seconds } = this.state
+      const gameRef = await this.props.firebase.game(id).get();
+
+      if (seconds > 0) {
+        this.setState({
+          seconds: this.calculateTime('seconds')
+        })
+      }
+      if (seconds === 0) {
+        if (minutes === 0 && hours === 0 && days === 0) {
+          this.sendEmail(gameRef, id);
+          game.set({
+            currentPlayerIndex: gameRef.data().currentPlayerIndex + 1,
+          }, { merge: true })
+          clearInterval(this.myInterval)
+        } else {
+          this.setState({
+            days: this.calculateTime('days'),
+            hours: this.calculateTime('hours'),
+            minutes: this.calculateTime('minutes'),
+            seconds: this.calculateTime('seconds')
+          })
+        }
+      }
+    }, 1000)
+  }
+
+  calculateTime(unitOfTime) {
+    switch(unitOfTime) {
+      case 'days':
+        return Math.floor((((this.state.timeTurnWillEnd - Math.floor(new Date().getTime() / 1000)) / 60) / 60)/ 24);
+      case 'hours':
+        return Math.floor(((this.state.timeTurnWillEnd - Math.floor(new Date().getTime() / 1000)) / 60) / 60) % 24;
+      case 'minutes':
+        return Math.floor((this.state.timeTurnWillEnd - Math.floor(new Date().getTime() / 1000)) / 60) % 60;
+      case 'seconds':
+        return (this.state.timeTurnWillEnd - Math.floor(new Date().getTime() / 1000)) % 60;
+      default:
+        throw new Error('invalid case')
+    }
+  }
+
+  sendEmail(snapshot, gameId) {
+    const emailType = (snapshot.data().currentPlayerIndex === snapshot.data().players.length) ? 'end' : 'turn';
+    fetch('/api/notify', {
+      method: 'POST',
+      headers: {
+        'Accept': 'application/x-www-form-urlencoded, multipart/form-data, text/plain',
+        'Content-Type': 'application/x-www-form-urlencoded'
+      },
+      body: `gameID=${gameId}&emailType=${emailType}`,
+    }).then((response) => {
+      console.log(response.text());
+    });
+  }
```

Comment:
if this code is copy-pasted from elsewhere, might be better to define in a Helper for code reuse